packages/php/compat-checker/src/Checks/WCCompatibility.php

```diff
@@ -0,0 +1,389 @@
+			$message        = sprintf(
+				/* translators: %1$s - Plugin Name, %2$s - activate WooCommerce link open, %3$s - activate WooCommerce link close. */
+				esc_html__( '%1$s requires WooCommerce to be activated. Please %2$sactivate WooCommerce%3$s.', 'woogrow-compat-checker' ),
+				'<strong>' . $plugin_name . '</strong>',
+				'<a href="' . esc_url( $activation_url ) . '">',
+				'</a>'
+			);
+			$this->add_admin_notice(
+				'woocommerce-not-activated',
+				'error',
+				$message
+			);
+		} else {
+			if ( ! current_user_can( 'install_plugins' ) ) {
+				return;
+			}
+
+			$install_url = wp_nonce_url( self_admin_url( 'update.php?action=install-plugin&plugin=woocommerce' ), 'install-plugin_woocommerce' );
+			$message     = sprintf(
+				/* translators: %1$s - Plugin Name, %2$s - install WooCommerce link open, %3$s - install WooCommerce link close. */
+				esc_html__( '%1$s requires WooCommerce to be installed and activated. Please %2$sinstall WooCommerce%3$s.', 'woogrow-compat-checker' ),
+				'<strong>' . $plugin_name . '</strong>',
+				'<a href="' . esc_url( $install_url ) . '">',
+				'</a>'
+			);
+
+			$this->add_admin_notice(
+				'woocommerce-not-installed',
+				'error',
+				$message
+			);
+		}
+	}
+
+	/**
+	 * Add notices for out of date WooCommerce.
+	 */
+	public function wc_out_of_date() {
+		if ( ! current_user_can( 'update_plugins' ) ) {
+			return;
+		}
+
+		$plugin_name         = $this->plugin_data['Name'];
+		$wc_version_required = $this->plugin_data['RequiresWC'];
+
+		$update_url = wp_nonce_url( self_admin_url( 'update.php?action=upgrade-plugin&plugin=' . self::WC_PLUGIN_FILE ), 'upgrade-plugin_' . self::WC_PLUGIN_FILE );
+		$message    = sprintf(
+			/* translators: %1$s - Plugin Name, %2$s - minimum WooCommerce version, %3$s - update WooCommerce link open, %4$s - update WooCommerce link close, %5$s - download minimum WooCommerce link open, %6$s - download minimum WooCommerce link close. */
+			esc_html__( '%1$s requires WooCommerce version %2$s or higher. Please %3$supdate WooCommerce%4$s to the latest version, or %5$sdownload the minimum required version &raquo;%6$s', 'woogrow-compat-checker' ),
+			'<strong>' . $plugin_name . '</strong>',
+			$wc_version_required,
+			'<a href="' . esc_url( $update_url ) . '">',
+			'</a>',
+			'<a href="' . esc_url( 'https://downloads.wordpress.org/plugin/woocommerce.' . $wc_version_required . '.zip' ) . '">',
+			'</a>'
+		);
+
+		$this->add_admin_notice(
+			'woocommerce-out-of-date',
+			'error',
+			$message
+		);
+	}
+
+	/**
+	 * Adds notice for untested WooCommerce.
+	 */
+	public function wc_untested() {
+		$plugin_name    = $this->plugin_data['Name'];
+		$wc_version     = $this->get_wc_version();
+		$plugin_version = $this->plugin_data['Version'];
+
+		$message = sprintf(
+			/* translators: %1$s - Plugin Name, %2$s - Plugin version, %3$s - WooCommerce version number */
+			esc_html__( '%1$s - %2$s is untested with WooCommerce %3$s.', 'woogrow-compat-checker' ),
```

Comment:
❓ Should the merchant be able to dismiss this error? Otherwise it just hangs out until the plugin is updated to support my WooCommerce version (good thing always on top of compatibility releases!!)

Reply:
I've made the warning notice dismissible in this commit: https://github.com/woocommerce/grow/pull/79/commits/c8c50e043763f275f1e5901ba2975b0aab2416f1 

The `is-dismissible` applies only on the current screen. To maintain the notice state, we'll have to store the notice state in the user meta data. Do you want me to implement that?

Reply:
Hmm ... I guess for WooCommerce it's not particularly critical because our extensions _should_ always be up to date with WooCommerce. We could either make it dismissible until the next WooCommerce update, or leave as is and then change it if we see that our extensions are falling behind on compatibility (+1 technical debt)